### PR TITLE
feat: added usb modem drivers

### PR DESCRIPTION
**Pull request by frezbo**
new extension that adds the required drivers to operate usb modems under Talos.

Based on #167

### CODE REVIEW

File: power/nut-client/pkg.yaml [hunk 1]
```diff
@@ -7,15 +7,15 @@ dependencies:
   - image: "{{ .PKGS_PREFIX }}/util-linux:{{ .BUILD_ARG_PKGS }}"
 steps:
   - sources:
-      - url: https://github.com/networkupstools/nut/releases/download/v{{ .NUT_VERSION }}/nut-{{ .NUT_VERSION }}.tar.gz
+      - url: https://github.com/networkupstools/nut/releases/download/v{{ .NUT_VERSION }}-signed/nut-{{ .NUT_VERSION }}.tar.gz
```
Comment (frezbo) [inline power/nut-client/pkg.yaml:10]: they re-released 2.8.0 and named it `-signed` for a stable one and normal one without `-signed` is now marked as pre-release for windows :shrug:

File: drivers/usb-modem/README.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ kern:    info: [2023-06-07T16:40:10.280291521Z]: qmi_wwan 6-1.2:1.4: cdc-wdm1: USB WDM device  
+```
+
+## Configurtion
+
+Talos currently does not have an OS level method of dialing up cellular modems, such as `ModemManager` or `uqmi`, so you will
+probably want to ignore it on the system level, pass the modems `/dev/` devices to a pod, and dial it up there using a tool such
+as `ModemManager` or `uqmi`.
+
+first, ignore the interface on the Talos level, so Talos doesn't try to bring it up:
```
Comment (Unix4ever) [inline drivers/usb-modem/README.md:75]:
```suggestion
First, ignore the interface on the Talos level, so Talos doesn't try to bring it up:
```

**Review by Unix4ever — APPROVED** [review]

**Comment by roeeklinger** [comment]
Because I'm the only one with the modem hardware, I just wanted to test this and report back that it works as expected:
```
talosctl -n 192.168.1.205 dmesg | grep "wwan"
192.168.1.205: kern:    info: [2023-06-09T16:25:27.634227565Z]: qmi_wwan 4-1.1:1.4: cdc-wdm0: USB WDM device
192.168.1.205: kern:    info: [2023-06-09T16:25:27.702080565Z]: qmi_wwan 4-1.1:1.4 wwan0: register 'qmi_wwan' at usb-0000:07:00.0-1.1, WWAN/QMI device, 92:19:d0:95:ab:c3
192.168.1.205: kern:    info: [2023-06-09T16:25:27.836000565Z]: qmi_wwan 6-1.2:1.4: cdc-wdm1: USB WDM device
192.168.1.205: kern:    info: [2023-06-09T16:25:27.916128565Z]: qmi_wwan 6-1.2:1.4 wwan1: register 'qmi_wwan' at usb-0000:1d:00.0-1.2, WWAN/QMI device, 92:19:d0:95:ab:c3
192.168.1.205: kern:    info: [2023-06-09T16:25:27.921210565Z]: usbcore: registered new interface driver qmi_wwan
192.168.1.205: user: warning: [2023-06-09T16:26:05.591797565Z]: [talos] request/renew failed {"component": "controller-runtime", "controller": "network.OperatorSpecController", "operator": "dhcp4", "error": "unable to receive an offer: got an error while the discovery request: no matching response packet received", "link": "wwan1"}
192.168.1.205: user: warning: [2023-06-09T16:26:05.889391565Z]: [talos] request/renew failed {"component": "controller-runtime", "controller": "network.OperatorSpecController", "operator": "dhcp4", "error": "unable to receive an offer: got an error while the discovery request: no matching response packet received", "link": "wwan0"}
```

The only issue I see is that `belkin_sa` was named in REAME file of the extensions, but it doesn't actually exist in pkgs (which is good I guess, because it seems like it is not really needed) so if you follow the README, the module is of course refusing to load since it doesn't exist:
```
talosctl -n 192.168.1.205 dmesg | grep "error loading"
192.168.1.205: user: warning: [2023-06-09T16:25:33.843951418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:25:34.294170418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:25:35.684935418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:25:37.150583418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:25:39.440275418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:25:42.553394418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:25:48.713454418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:25:52.203308418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:26:04.017169418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:26:12.980095418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
192.168.1.205: user: warning: [2023-06-09T16:26:32.727795418Z]: [talos] controller failed {"component": "controller-runtime", "controller": "runtime.KernelModuleSpecController", "error": "error loading module \x5c"belkin_sa\x5c": module not found"}
```

I will send a PR now removing it from the list

**Pull request merged**